### ISSUE TITLE
when using #!/command/with-contenv sh ifelse and /run/s6/container_environment not found

### ISSUE THREAD

**Issue by Impakt** [just-containers/s6-overlay#524]
Hello,
I built a plain alpine 3.17.3 image and installed the  noarch and platform specific overlays. They installed ok but when running a script with `#!/command/with-contenv sh` i get this error `execlineb: fatal: unable to exec ifelse: No such file or directory`.

If i add /command to the PATH, i then get `s6-envdir: fatal: unable to envdir /run/s6/container_environment: No such file or directory`

Do I need to install the symlinks and create this dir for it to work?

**Comment by skarnet** [comment]
It sounds like you have installed s6-overlay but are not running it. Is `/init` your ENTRYPOINT?

**Comment by Impakt** [comment]
yes it is. The service im trying to run is fluentd, which is at `etc/s6-overlay/s6-rc.d/fluentd/run`, and has 

```bash
#!/command/with-contenv sh
exec /bin/fluentd-entrypoint.sh
```
as the content.

The container will start then stop instantly. These errors happen when i try and execute the `run` file from the shell

```
s6-rc: info: service fluentd: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service fluentd successfully started
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service fluentd: stopping
s6-rc: info: service fluentd successfully stopped
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by skarnet** [comment]
The actual error messages should appear in your log, but they don't seem to be here. Do you have a dedicated logger for fluentd? If so, what error messages appear there?

**Comment by Impakt** [comment]
Again i apologise for creating another issue. Turns out the service was quiiting without logging anything. I created a new Dockerfile from the container in question, added an entrypoint to `tail -f /dev/null`, then sh'd in to diagnose the issue. I had to add `/command` to the path for it to work though, and i created the `/run/s6/container_environment` dir for good measure.  Appreciate the support @skarnet !
